src/cli_shared/mod.rs

```diff
@@ -11,9 +11,11 @@ pub use mimalloc;
 #[cfg(feature = "jemalloc")]
 pub use tikv_jemallocator;
 
+use crate::networks::NetworkChain;
+
 /// Gets chain data directory
-pub fn chain_path(config: &crate::cli_shared::cli::Config) -> PathBuf {
-    PathBuf::from(&config.client.data_dir).join(config.chain.network.to_string())
+pub fn chain_path(network: &NetworkChain, config: &crate::cli_shared::cli::Config) -> PathBuf {
+    PathBuf::from(&config.client.data_dir).join(network.to_string())
```

Comment:
How is `network` different from `config.chain`?

Reply:
CLI `--chain` args has higher priority than `config.chain` and their value may disagree, I guess I can override `config.chain` at the very beginning instead.

Reply:
Fixed